malta: add support for 6.18 kernel version

namiltd opened the pull request:
Kernel 6.18-rc7 was released on November 23, 2025, so I decided to add support for it. It will probably be an LTS release.

Updated to Kernel 6.18.20 (migrated from 6.12.77).

Known issues:
 - ~crypto.mk need to be refreshed for 6.18~
 - ~hack patches not converted yet:~
   ~- 301-02-mips-replace-mlong-calls-with-mno-long-calls-if-poss.patch~
   ~- 800-GPIO-add-named-gpio-exports.patch~
 - ~proper config-6.18 files for generic and malta haven't been created yet~

namiltd commented:
First successful compilation and run of version 6.18-rc7:

<img width="969" height="1109" alt="obraz" src="https://github.com/user-attachments/assets/4cfcda81-c622-4681-9f73-d4df509ef4e2" />

`[29 lis 2025, 17:40:19 UTC] kern.notice: [    0.000000] Linux version 6.18 (root@devuan) (mipsel-openwrt-linux-musl-gcc (OpenWrt GCC 14.3.0 r32020-1c136355e0) 14.3.0, GNU ld (GNU Binutils) 2.44) #0 SMP Fri Nov 28 16:58:56 2025
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] earlycon: uart8250 at I/O port 0x3f8 (options '38400n8')
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] printk: legacy bootconsole [uart8250] enabled
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] CPU0 revision is: 00019300 (MIPS 24Kc)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] MIPS: machine is mti,malta
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Software DMA cache coherency enabled
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] OF: reserved mem: Reserved memory: No reserved-memory node in the DT
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Primary instruction cache 2kB, VIPT, 2-way, linesize 16 bytes.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Primary data cache 2kB, 2-way, VIPT, no aliases, linesize 16 bytes
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Zone ranges:
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000]   DMA      [mem 0x0000000000000000-0x0000000000ffffff]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000]   Normal   [mem 0x0000000001000000-0x000000000fffffff]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000]   HighMem  empty
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Movable zone start for each node
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Early memory node ranges
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000]   node   0: [mem 0x0000000000000000-0x000000000fffffff]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000000fffffff]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] percpu: Embedded 13 pages/cpu s24464 r8192 d20592 u53248
[29 lis 2025, 17:40:19 UTC] kern.debug: [    0.000000] pcpu-alloc: s24464 r8192 d20592 u53248 alloc=13*4096
[29 lis 2025, 17:40:19 UTC] kern.debug: [    0.000000] pcpu-alloc: [0] 0
[29 lis 2025, 17:40:19 UTC] kern.notice: [    0.000000] Kernel command line: root=/dev/sda console=ttyS0 rootfstype=ext4
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] printk: log buffer data + meta data: 131072 + 409600 = 540672 bytes
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Dentry cache hash table entries: 32768 (order: 5, 131072 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Inode-cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Writing ErrCtl register=00000000
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Readback ErrCtl register=00000000
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 65536
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] rcu: Hierarchical RCU implementation.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] rcu: 	RCU restricting CPUs from NR_CPUS=2 to nr_cpu_ids=1.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] 	Tracing variant of Tasks RCU enabled.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=1
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] RCU Tasks Trace: Setting shift to 0 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=1.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] NR_IRQS: 256
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[29 lis 2025, 17:40:19 UTC] kern.warn: [    0.000000] CPU frequency 333.32 MHz
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 11468075693 ns
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.000502] sched_clock: 32 bits at 167MHz, resolution 6ns, wraps every 12885478396ns
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.023582] Console: colour dummy device 80x25
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.031150] Calibrating delay loop...
[29 lis 2025, 17:40:19 UTC] kern.debug: [    0.049974] spurious 8259A interrupt: IRQ7.
[29 lis 2025, 17:40:19 UTC] kern.warn: [    0.139950] 580.40 BogoMIPS (lpj=2902016)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.142945] pid_max: default: 32768 minimum: 301
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.170935] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.175026] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.330568] rcu: Hierarchical SRCU implementation.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.331403] rcu: 	Max phase no-delay instances is 1000.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.346855] smp: Bringing up secondary CPUs ...
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.348045] smp: Brought up 1 node, 1 CPU
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.368412] Memory: 246124K/262144K available (8541K kernel code, 647K rwdata, 1696K rodata, 308K init, 249K bss, 15048K reserved, 0K cma-reserved, 0K highmem)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.407741] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.411249] posixtimers hash table entries: 512 (order: 0, 4096 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.412873] futex hash table entries: 256 (16384 bytes on 1 NUMA nodes, total 16 KiB, linear).
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.446882] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[29 lis 2025, 17:40:19 UTC] kern.debug: [    0.467464] FPU Affinity set after 11600 emulations
[29 lis 2025, 17:40:19 UTC] kern.notice: [    0.531228] SCSI subsystem initialized
[29 lis 2025, 17:40:19 UTC] kern.debug: [    0.534482] libata version 3.00 loaded.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.571622] PCI host bridge to bus 0000:00
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.573338] pci_bus 0000:00: root bus resource [mem 0x10000000-0x17ffffff]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.576703] pci_bus 0000:00: root bus resource [io  0x0000-0x1fffff]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.577811] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.582451] pci 0000:00:00.0: [11ab:4620] type 00 class 0x060000 conventional PCI endpoint
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.585249] pci 0000:00:00.0: [Firmware Bug]: BAR 1: invalid; can't size
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.586079] pci 0000:00:00.0: [Firmware Bug]: BAR 2: invalid; can't size
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.588074] pci 0000:00:00.0: [Firmware Bug]: BAR 3: invalid; can't size
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.592674] pci 0000:00:00.0: [Firmware Bug]: BAR 4: invalid; can't size
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.593802] pci 0000:00:00.0: [Firmware Bug]: BAR 5: invalid; can't size
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.601150] pci 0000:00:0a.0: [8086:7110] type 00 class 0x060100 conventional PCI endpoint
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.603782] pci 0000:00:0a.1: [8086:7111] type 00 class 0x010180 conventional PCI endpoint
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.607503] pci 0000:00:0a.1: BAR 4 [io  0x0000-0x000f]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.610921] pci 0000:00:0a.1: BAR 0 [io  0x01f0-0x01f7]: legacy IDE quirk
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.611751] pci 0000:00:0a.1: BAR 1 [io  0x03f6]: legacy IDE quirk
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.612592] pci 0000:00:0a.1: BAR 2 [io  0x0170-0x0177]: legacy IDE quirk
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.613326] pci 0000:00:0a.1: BAR 3 [io  0x0376]: legacy IDE quirk
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.615020] pci 0000:00:0a.2: [8086:7112] type 00 class 0x0c0300 conventional PCI endpoint
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.616084] pci 0000:00:0a.2: BAR 4 [io  0x0000-0x001f]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.617630] pci 0000:00:0a.3: [8086:7113] type 00 class 0x068000 conventional PCI endpoint
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.620133] pci 0000:00:0a.3: quirk: [io  0x1000-0x103f] claimed by PIIX4 ACPI
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.625786] pci 0000:00:0a.3: quirk: [io  0x1100-0x110f] claimed by PIIX4 SMB
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.628119] pci 0000:00:12.0: [1013:00b8] type 00 class 0x030000 conventional PCI endpoint
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.629368] pci 0000:00:12.0: BAR 0 [mem 0x00000000-0x01ffffff pref]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.630235] pci 0000:00:12.0: BAR 1 [mem 0x00000000-0x00000fff]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.630918] pci 0000:00:12.0: ROM [mem 0x00000000-0x0000ffff pref]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.632182] pci 0000:00:13.0: [1022:2000] type 00 class 0x020000 conventional PCI endpoint
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.634158] pci 0000:00:13.0: BAR 0 [io  0x0000-0x001f]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.634847] pci 0000:00:13.0: BAR 1 [mem 0x00000000-0x0000001f]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.635500] pci 0000:00:13.0: ROM [mem 0x00000000-0x0003ffff pref]
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.639966] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.646939] pci 0000:00:12.0: BAR 0 [mem 0x10000000-0x11ffffff pref]: assigned
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.648543] pci 0000:00:13.0: ROM [mem 0x12000000-0x1203ffff pref]: assigned
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.649509] pci 0000:00:12.0: ROM [mem 0x12040000-0x1204ffff pref]: assigned
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.650342] pci 0000:00:12.0: BAR 1 [mem 0x12050000-0x12050fff]: assigned
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.651203] pci 0000:00:0a.2: BAR 4 [io  0x1040-0x105f]: assigned
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.651965] pci 0000:00:13.0: BAR 0 [io  0x1060-0x107f]: assigned
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.652730] pci 0000:00:13.0: BAR 1 [mem 0x12051000-0x1205101f]: assigned
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.653585] pci 0000:00:0a.1: BAR 4 [io  0x1080-0x108f]: assigned
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.664457] clocksource: Switched to clocksource MIPS
[29 lis 2025, 17:40:19 UTC] kern.notice: [    0.693229] VFS: Disk quotas dquot_6.6.0
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.694732] VFS: Dquot-cache hash table entries: 1024 (order 0, 4096 bytes)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.727201] NET: Registered PF_INET protocol family
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.732674] IP idents hash table entries: 4096 (order: 3, 32768 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.748293] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.749796] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.750925] TCP established hash table entries: 2048 (order: 1, 8192 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.752327] TCP bind hash table entries: 2048 (order: 3, 32768 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.755459] TCP: Hash tables configured (established 2048 bind 2048)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.774026] MPTCP token hash table entries: 256 (order: 0, 4096 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.779378] UDP hash table entries: 256 (order: 2, 14336 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.780797] UDP-Lite hash table entries: 256 (order: 2, 14336 bytes, linear)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.789157] NET: Registered PF_UNIX/PF_LOCAL protocol family
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.793790] PCI: CLS 0 bytes, default 16
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.835913] workingset: timestamp_bits=14 max_order=16 bucket_order=2
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.871719] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.876004] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.917713] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 252)
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.944129] Serial: 8250/16550 driver, 4 ports, IRQ sharing disabled
[29 lis 2025, 17:40:19 UTC] kern.info: [    0.963422] printk: legacy console [ttyS0] disabled
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.005651] serial8250.0: ttyS0 at I/O 0x3f8 (irq = 4, base_baud = 115200) is a 16550A
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.009996] serial8250.0: ttyS0 extra baud rates supported: 230400, 460800
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.014896] printk: legacy console [ttyS0] enabled
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.018644] printk: legacy bootconsole [uart8250] disabled
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.085528] serial8250.0: ttyS1 at I/O 0x2f8 (irq = 3, base_baud = 115200) is a 16550A
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.090093] serial8250.0: ttyS1 extra baud rates supported: 230400, 460800
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.122769] serial8250.0: ttyS2 at MMIO 0x1f000900 (irq = 20, base_baud = 230400) is a 16550A
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.222202] brd: module loaded
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.269486] loop: module loaded
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.275731] ata_piix 0000:00:0a.1: enabling device (0000 -> 0001)
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.316603] scsi host0: ata_piix
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.329881] scsi host1: ata_piix
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.332506] ata1: PATA max UDMA/33 cmd 0x1f0 ctl 0x3f6 bmdma 0x1080 irq 14 lpm-pol 0
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.333610] ata2: PATA max UDMA/33 cmd 0x170 ctl 0x376 bmdma 0x1088 irq 15 lpm-pol 0
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.347358] mousedev: PS/2 mouse device common for all mice
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.359645] rtc_cmos 70.rtc: registered as rtc0
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.362771] rtc_cmos 70.rtc: setting system clock to 2025-11-29T17:39:37 UTC (1764437977)
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.365397] rtc_cmos 70.rtc: alarms up to one day, 242 bytes nvram
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.380945] NET: Registered PF_INET6 protocol family
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.426199] Segment Routing with IPv6
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.427450] In-situ OAM (IOAM) with IPv6
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.432276] NET: Registered PF_PACKET protocol family
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.434842] 8021q: 802.1Q VLAN Support v1.8
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.563376] ata2: found unknown device (class 0)
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.576109] ata1: found unknown device (class 0)
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.591895] ata1.00: ATA-7: QEMU HARDDISK, 2.5+, max UDMA/100
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.593207] ata1.00: 212992 sectors, multi 16: LBA48
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.631611] ata2.00: ATAPI: QEMU DVD-ROM, 2.5+, max UDMA/100
[29 lis 2025, 17:40:19 UTC] kern.notice: [    1.678041] scsi 0:0:0:0: Direct-Access     ATA      QEMU HARDDISK    2.5+ PQ: 0 ANSI: 5
[29 lis 2025, 17:40:19 UTC] kern.notice: [    1.712603] sd 0:0:0:0: [sda] 212992 512-byte logical blocks: (109 MB/104 MiB)
[29 lis 2025, 17:40:19 UTC] kern.notice: [    1.715785] sd 0:0:0:0: [sda] Write Protect is off
[29 lis 2025, 17:40:19 UTC] kern.debug: [    1.718090] sd 0:0:0:0: [sda] Mode Sense: 00 3a 00 00
[29 lis 2025, 17:40:19 UTC] kern.notice: [    1.719711] sd 0:0:0:0: [sda] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.725641] sd 0:0:0:0: [sda] Preferred minimum I/O size 512 bytes
[29 lis 2025, 17:40:19 UTC] kern.notice: [    1.742315] scsi 1:0:0:0: CD-ROM            QEMU     QEMU DVD-ROM     2.5+ PQ: 0 ANSI: 5
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.764340] clk: Disabling unused clocks
[29 lis 2025, 17:40:19 UTC] kern.notice: [    1.786894] sd 0:0:0:0: [sda] Attached SCSI disk
[29 lis 2025, 17:40:19 UTC] kern.warn: [    1.839228] check access for rdinit=/init failed: -2, ignoring
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.929733] EXT4-fs (sda): mounted filesystem ff313567-e9f1-5a5d-9895-3ba130b4a864 ro with ordered data mode. Quota mode: none.
[29 lis 2025, 17:40:19 UTC] kern.info: [    1.939903] VFS: Mounted root (ext4 filesystem) readonly on device 8:0.
[29 lis 2025, 17:40:19 UTC] kern.info: [    2.040616] Freeing unused kernel image (initmem) memory: 308K
[29 lis 2025, 17:40:19 UTC] kern.warn: [    2.041529] This architecture does not have kernel memory protection.
[29 lis 2025, 17:40:19 UTC] kern.info: [    2.045599] Run /sbin/init as init process
[29 lis 2025, 17:40:19 UTC] kern.debug: [    2.046125]   with arguments:
[29 lis 2025, 17:40:19 UTC] kern.debug: [    2.046191]     /sbin/init
[29 lis 2025, 17:40:19 UTC] kern.debug: [    2.046255]   with environment:
[29 lis 2025, 17:40:19 UTC] kern.debug: [    2.046305]     HOME=/
[29 lis 2025, 17:40:19 UTC] kern.debug: [    2.046373]     TERM=linux
[29 lis 2025, 17:40:19 UTC] user.info: [    2.738456] init: Console is alive
[29 lis 2025, 17:40:19 UTC] user.info: [    3.000796] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[29 lis 2025, 17:40:19 UTC] user.info: [    3.047160] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[29 lis 2025, 17:40:19 UTC] user.info: [    3.078908] init: - preinit -
[29 lis 2025, 17:40:19 UTC] kern.warn: [    5.376647] hrtimer: interrupt took 56325624 ns
[29 lis 2025, 17:40:19 UTC] kern.notice: [   12.174382] random: crng init done
[29 lis 2025, 17:40:19 UTC] user.notice: [   20.314301] mount_root: mounting /dev/root with options
[29 lis 2025, 17:40:19 UTC] kern.info: [   20.349719] EXT4-fs (sda): re-mounted ff313567-e9f1-5a5d-9895-3ba130b4a864 r/w.
[29 lis 2025, 17:40:19 UTC] user.warn: [   20.369106] urandom-seed: Seed file not found (/etc/urandom.seed)
[29 lis 2025, 17:40:19 UTC] user.info: [   20.714182] procd: - early -
[29 lis 2025, 17:40:19 UTC] user.info: [   22.184733] procd: - ubus -
[29 lis 2025, 17:40:19 UTC] user.info: [   22.454412] procd: - init -
[29 lis 2025, 17:40:19 UTC] user.info: [   30.691675] kmodloader: loading kernel modules from /etc/modules.d/*
[29 lis 2025, 17:40:19 UTC] user.info: [   31.070962] urngd: v1.0.2 started.
[29 lis 2025, 17:40:19 UTC] kern.info: [   34.045089] pcnet32 0000:00:13.0: enabling device (0000 -> 0003)
[29 lis 2025, 17:40:19 UTC] kern.info: [   34.055043] pcnet32: PCnet/PCI II 79C970A at 0x1060, 52:54:00:12:34:56 assigned IRQ 10
[29 lis 2025, 17:40:19 UTC] kern.info: [   34.080381] pcnet32: eth0: registered as PCnet/PCI II 79C970A
[29 lis 2025, 17:40:19 UTC] kern.info: [   34.109315] pcnet32: 1 cards_found
[29 lis 2025, 17:40:19 UTC] user.info: [   34.691964] kmodloader: done loading kernel modules from /etc/modules.d/*
[29 lis 2025, 17:40:25 UTC] user.notice: dnsmasq: DNS rebinding protection is active, will discard upstream RFC1918 responses!
[29 lis 2025, 17:40:25 UTC] user.notice: dnsmasq: Allowing 127.0.0.0/8 responses
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: started, version 2.91 cachesize 1000
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: DNS service limited to local subnets
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-nftset no-auth no-DNSSEC no-ID loop-detect inotify dumpfile
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: UBus support enabled: connected to system bus
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for test
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for onion
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for localhost
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for local
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for invalid
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for bind
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for lan
[29 lis 2025, 17:40:28 UTC] daemon.warn: dnsmasq[1]: no servers found in /tmp/resolv.conf.d/resolv.conf.auto, will retry
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: read /etc/hosts - 12 names
[29 lis 2025, 17:40:28 UTC] daemon.info: dnsmasq[1]: read /tmp/hosts/dhcp.cfg01411c - 0 names
[29 lis 2025, 17:40:31 UTC] authpriv.info: dropbear[1151]: Not backgrounding
[29 lis 2025, 17:40:39 UTC] daemon.notice: wpa_supplicant[1276]: Successfully initialized wpa_supplicant
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: vrf
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: bonding
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: 8021ad
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: 8021q
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: macvlan
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: veth
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: bridge
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: Network device
[29 lis 2025, 17:40:40 UTC] user.notice: : Added device handler type: tunnel
[29 lis 2025, 17:40:48 UTC] daemon.notice: wpa_supplicant[1276]: Set MLD config: [ ]
[29 lis 2025, 17:40:48 UTC] daemon.notice: hostapd: Set MLD config: [ ]
[29 lis 2025, 17:40:48 UTC] daemon.notice: hostapd: Reload all interfaces
[29 lis 2025, 17:40:48 UTC] kern.info: [   72.716838] pcnet32 0000:00:13.0 eth0: link up
[29 lis 2025, 17:40:48 UTC] kern.info: [   72.768518] br-lan: port 1(eth0) entered blocking state
[29 lis 2025, 17:40:48 UTC] kern.info: [   72.769717] br-lan: port 1(eth0) entered disabled state
[29 lis 2025, 17:40:48 UTC] kern.info: [   72.774319] pcnet32 0000:00:13.0 eth0: entered allmulticast mode
[29 lis 2025, 17:40:48 UTC] kern.info: [   72.781863] pcnet32 0000:00:13.0 eth0: entered promiscuous mode
[29 lis 2025, 17:40:48 UTC] kern.info: [   72.948561] br-lan: port 1(eth0) entered blocking state
[29 lis 2025, 17:40:48 UTC] kern.info: [   72.949589] br-lan: port 1(eth0) entered forwarding state
[29 lis 2025, 17:40:48 UTC] daemon.notice: netifd: Interface 'lan' is enabled
[29 lis 2025, 17:40:48 UTC] daemon.notice: netifd: Interface 'lan' is setting up now
[29 lis 2025, 17:40:48 UTC] daemon.notice: netifd: Interface 'lan' is now up
[29 lis 2025, 17:40:48 UTC] daemon.notice: netifd: bridge 'br-lan' link is up
[29 lis 2025, 17:40:48 UTC] daemon.notice: netifd: Interface 'lan' has link connectivity
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: Interface 'loopback' is enabled
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: Interface 'loopback' is setting up now
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: Interface 'loopback' is now up
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: bridge 'br-lan' link is down
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: Interface 'lan' has link connectivity loss
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: Network device 'eth0' link is up
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: bridge 'br-lan' link is up
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: Interface 'lan' has link connectivity
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: Network device 'lo' link is up
[29 lis 2025, 17:40:49 UTC] daemon.notice: netifd: Interface 'loopback' has link connectivity
[29 lis 2025, 17:40:49 UTC] daemon.notice: procd: /etc/rc.d/S50uhttpd: 4+0 records in
[29 lis 2025, 17:40:49 UTC] daemon.notice: procd: /etc/rc.d/S50uhttpd: 4+0 records out
[29 lis 2025, 17:40:54 UTC] user.notice: firewall: Reloading firewall due to ifup of lan (br-lan)
[29 lis 2025, 17:40:55 UTC] user.notice: ucitrack: Setting up /etc/config/luci-splash reload dependency on /etc/config/firewall
[29 lis 2025, 17:40:55 UTC] user.notice: ucitrack: Setting up /etc/config/qos reload dependency on /etc/config/firewall
[29 lis 2025, 17:40:56 UTC] user.notice: ucitrack: Setting up /etc/config/miniupnpd reload dependency on /etc/config/firewall
[29 lis 2025, 17:40:58 UTC] user.notice: ucitrack: Setting up /etc/config/odhcpd reload dependency on /etc/config/dhcp
[29 lis 2025, 17:40:59 UTC] user.notice: ucitrack: Setting up /etc/config/dhcp reload dependency on /etc/config/network
[29 lis 2025, 17:41:00 UTC] user.notice: ucitrack: Setting up /etc/config/network reload dependency on /etc/config/wireless
[29 lis 2025, 17:41:02 UTC] user.notice: ucitrack: Setting up non-init /etc/config/fstab reload handler: /sbin/block mount
[29 lis 2025, 17:41:03 UTC] user.notice: ucitrack: Setting up /etc/config/system reload trigger for non-procd /etc/init.d/led
[29 lis 2025, 17:41:05 UTC] user.notice: ucitrack: Setting up /etc/config/luci_statistics reload dependency on /etc/config/system
[29 lis 2025, 17:41:05 UTC] user.notice: ucitrack: Setting up /etc/config/dhcp reload dependency on /etc/config/system
[29 lis 2025, 17:41:06 UTC] daemon.warn: odhcpd[1444]: No default route present, setting ra_lifetime to 0!
[29 lis 2025, 17:41:06 UTC] daemon.warn: odhcpd[1444]: rfc9096: br-lan: piofile updated
[29 lis 2025, 17:41:07 UTC] daemon.info: dnsmasq[1]: exiting on receipt of SIGTERM
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: started, version 2.91 cachesize 1000
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: DNS service limited to local subnets
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-nftset no-auth no-DNSSEC no-ID loop-detect inotify dumpfile
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: UBus support enabled: connected to system bus
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq-dhcp[1]: DHCP, IP range 192.168.10.100 -- 192.168.10.249, lease time 12h
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for test
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for onion
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for localhost
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for local
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for invalid
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for bind
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: using only locally-known addresses for lan
[29 lis 2025, 17:41:09 UTC] daemon.warn: dnsmasq[1]: no servers found in /tmp/resolv.conf.d/resolv.conf.auto, will retry
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: read /etc/hosts - 12 names
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq[1]: read /tmp/hosts/dhcp.cfg01411c - 4 names
[29 lis 2025, 17:41:09 UTC] daemon.info: dnsmasq-dhcp[1]: read /etc/ethers - 0 addresses
[29 lis 2025, 17:41:16 UTC] daemon.info: procd: - init complete -
[29 lis 2025, 17:41:17 UTC] daemon.info: urandom_seed[2256]: Seed saved (/etc/urandom.seed)
[29 lis 2025, 17:41:22 UTC] daemon.warn: odhcpd[1444]: No default route present, setting ra_lifetime to 0!
[29 lis 2025, 17:41:38 UTC] daemon.warn: odhcpd[1444]: No default route present, setting ra_lifetime to 0!
[29 lis 2025, 17:43:05 UTC] authpriv.info: dispatcher.uc: luci: accepted login on / for root from 192.168.10.111
[29 lis 2025, 17:43:36 UTC] auth.err: passwd: password for root changed by root
[29 lis 2025, 17:49:59 UTC] daemon.warn: odhcpd[1444]: No default route present, setting ra_lifetime to 0!`

namiltd committed:
malta: 6.18: update kernel symbols

Remove symbols no longer present in version 6.18,
add new 6.18 kernel symbols.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/20965
Signed-off-by: Robert Marko <robimarko@gmail.com>

namiltd committed:
malta: enable 6.18 testing kernel

The malta now supports 6.18 kernel as testing.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/20965
Signed-off-by: Robert Marko <robimarko@gmail.com>

namiltd commented:
So far I have only built images locally.

graysky2 commented:
Right but have you enabled the buildbot option in menuconfig?
https://proton.me/mail/home

-------- Original Message --------
On Thursday, 12/11/25 at 17:42 Mieczyslaw Nalewaj ***@***.***> wrote:

> namiltd left a comment [(openwrt/openwrt#20965)](https://github.com/openwrt/openwrt/pull/20965#issuecomment-3644094393)
>
> So far I have only built images locally.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/openwrt/openwrt/pull/20965#issuecomment-3644094393), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AAECSJVASMMJEUZ5S3EDBBL4BHXNLAVCNFSM6AAAAACNNIIXQKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMNBUGA4TIMZZGM).
> You are receiving this because you commented.Message ID: ***@***.***>

namiltd commented:
Not yet.

graysky2 commented:
@namiltd - I noticed that you are adapting this PR working on a number of module files that I am. How can we avoid duplication of efforts?

graysky2 commented on package/kernel/linux/modules/netdevices.mk:
```diff
@@ -120,6 +120,38 @@ endef
+define KernelPackage/libie-fwlog
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Intel Ethernet library fw log
+  DEPENDS:=@LINUX_6_18 +kmod-libie
```
Never mind ... I see this is due to the inclusion of kmod-ixgbe in aae3a8a254275f8be5c45d766ac7b5afb82a9fe6

namiltd commented:
Why do errors appear so often in CI?:
```
 13150K .......... .......... .......... .......... .......... 99%  276M 0s
 13200K .......... .......... .......... .......... .......... 99%  246M 0s
 13250K .......... .......... .......... .......... .......... 99%  237M 0s
 13300K .......... .......                                    100%  235M=0.1s

2025-12-24 11:31:50 (116 MB/s) - ‘sha256sums’ saved [13637008/13637008]

gpg: assuming signed data in 'sha256sums'
gpg: Signature made Tue 23 Dec 2025 10:15:21 PM UTC
gpg:                using EDDSA key 92C561DE55AE6552F3C736B82B0151090606D1D9
gpg: BAD signature from "OpenWrt Build System (Nitrokey3) <contact@openwrt.org>" [unknown]
```
Apart from synchronization with main, my code has remained practically unchanged.

graysky2 commented on package/kernel/linux/modules/lib.mk:
```diff
@@ -86,7 +91,7 @@ define KernelPackage/lib-crc32c
   TITLE:=CRC32 support
   KCONFIG:=CONFIG_LIBCRC32C
-  DEPENDS:=+kmod-crypto-crc32c
+  DEPENDS:=@!LINUX_6_18 +kmod-crypto-crc32c
```
Is using `ifndef CONFIG_LINUX_6_18`  better?

namiltd commented:
Could someone with write access please re-run CI job https://github.com/openwrt/openwrt/actions/runs/20519270255/job/58951876682?  Thanks!

plappermaul commented:
As said above. The pending-6.18/487-mtd-spinand-Add-support-for-Etron-EM73D044VCx.patch copy breaks our realtek. @jonasjelonek made a temporary fix https://github.com/openwrt/openwrt/pull/21181/commits/755b8ea9d5a1bf46fcd71674740004717ebdb341

As a proper solution ... Via what PR do we want to take care of that?

namiltd commented:
@plappermaul Please paste the compilation error log.

namiltd commented:
@plappermaul Replace [487-mtd-spinand-Add-support-for-Etron-EM73D044VCx.patch](https://github.com/user-attachments/files/24354843/487-mtd-spinand-Add-support-for-Etron-EM73D044VCx.patch) and compile again.

jonasjelonek commented:
Compiles now, thanks!

plappermaul reviewed (approved):
Based on these patches the Realtek target already did its homework. See https://github.com/openwrt/openwrt/pull/21181 If possible we would like to aovid further pre-6.18 commits and rebase our 6.18 PR each time. So from our side this is good to go.

graysky2 commented:
@namiltd - are you running this on your test or production device? Any issues?

hauke commented on package/kernel/linux/modules/netsupport.mk:
```diff
@@ -1233,6 +1233,7 @@ define KernelPackage/sctp
      CONFIG_SCTP_COOKIE_HMAC_MD5=y \
      CONFIG_SCTP_DEFAULT_COOKIE_HMAC_NONE=n \
      CONFIG_SCTP_DEFAULT_COOKIE_HMAC_SHA1=n \
+     CONFIG_SCTP_DEFAULT_COOKIE_HMAC_SHA256=n@ge6.18 \
```
You can remove the condition here. The older kernel will just ignore the unknown config option.

hauke commented on target/linux/generic/hack-6.18/940-kbuild-extract-modules-builtin-modinfo-from-vmlinux_o.patch:
Please add a patch description to the patch file itself too.

graysky2 commented:
With all the changes you've been pushing of late, what if anything do I need to propagate in  https://github.com/openwrt/openwrt/pull/21078

tk154 commented:
> @tk154 Please attach the configuration in which this occurred.

I attached my [.config](https://github.com/user-attachments/files/24994679/config.txt). This seems to happen as soon as I enable `CONFIG_KERNEL_KPROBES`.

gingernut1978 commented:
Getting this compile error on latest

```
net/sched/sch_cake.o net/sched/sch_cake.c  
In file included from ./include/linux/module.h:23,
                 from net/sched/sch_cake.c:53:
net/sched/sch_cake.c:3381:18: error: expected ',' or ';' before 'NET_SCHED_INTERNAL'
 3381 | MODULE_IMPORT_NS(NET_SCHED_INTERNAL);
      |                  ^~~~~~~~~~~~~~~~~~
./include/linux/moduleparam.h:41:61: note: in definition of macro 'MODULE_INFO'
   41 |                 = __MODULE_INFO_PREFIX __stringify(tag) "=" info
      |                                                             ^~~~
net/sched/sch_cake.c:3381:1: note: in expansion of macro 'MODULE_IMPORT_NS'
 3381 | MODULE_IMPORT_NS(NET_SCHED_INTERNAL);
```

gingernut1978 commented:
@namiltd fixed.

Thank you

jonasjelonek reviewed (approved):
would be great to have this merged, realtek target is running like a charm with this as a foundation

andrewjlamarche reviewed (approved):
LGTM.

Tested on SBE1V1K (qualcommbe).

namiltd commented:
@andrewjlamarche If you have any problems with the batman-adv package, please check https://github.com/openwrt/routing,

namiltd commented:
For kernel 6.18 you need to disable the kmod-lib-crc32c package (in makefile of the batman-adv package).

namiltd committed:
kernel/malta: create files for v6.18 (from v6.12)

This is an automatically generated commit.

When doing `git bisect`, consider `git bisect --skip`.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/20965
Signed-off-by: Robert Marko <robimarko@gmail.com>

namiltd committed:
kernel/malta: restore files for v6.12

This is an automatically generated commit which aids following Kernel patch
history, as git will see the move and copy as a rename thus defeating the
purpose.

For the original discussion see:
https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/20965
Signed-off-by: Robert Marko <robimarko@gmail.com>

namiltd commented:
Ping @robimarko
